internal/config/config.go

```diff
@@ -407,7 +424,7 @@ func New(fileHandler file.Handler, name string, fetcher attestationconfigapi.Fet
 	// Read secrets from env-vars.
 	clientSecretValue := os.Getenv(constants.EnvVarAzureClientSecretValue)
 	if clientSecretValue != "" && c.Provider.Azure != nil {
-		c.Provider.Azure.ClientSecretValue = clientSecretValue
+		return nil, fmt.Errorf("unsupported environment variable: %s : %w", constants.EnvVarAzureClientSecretValue, UnsupportedAppRegistrationError{})
```

Comment:
We shouldn't error in case an env variable is set that we no longer support. Informing the user about the deprecation seems like a nice idea though. A warning should be enough imo.